Make it work with multiple sites

oligot opened the pull request:
The idea is to have a single node instance handling multiple sites
where each site has it's own directory structure.
This is especially useful for development where you don't want to
install and setup a full web server (like Apache or Nginx).

In this setup, you want to pass the sites to chokidar so that when
files change, the page is hot reloaded.

Suppose we have two sites:
- one
- two

When we change the file _one/app.js_, the emitted event will receive
the full path, i.e. _one/app.js_, while the hot reloader expect to
have a path without the prefix (in this case, _one/_).

This commit remove the prefix from the path so that the hot reloader
works as expected.

I've also created a small example to show how this works:
https://github.com/oligot/jspm-hot-reloader-multiple

capaj commented:
oh yeah, I see now. There is a problem though, because if you're using it with html5 mode and the path matches to a url of a module, by replacing that part you malform the path to the module.
We could replace it if a hashbang URL is detected. Would that work for you? Do your sites use hasbang history or html5 history pushtate?

oligot commented:
We don't use hashbang nor html5 history.
Maybe the method used to differentiate sites (plain old url, hashbang or html5 history) could be a parameter of JspmHotReloader class constructor ?

oligot commented:
I've updated the PR to add a new optional parameter `transform` to the JspmHotReloader constructor.
I've also updated the small example to show how one can use it
https://github.com/oligot/jspm-hot-reloader-multiple/blob/master/hot-reload.js#L7-L9

capaj commented:
@oligot this seems allright. I will merge this, but I want to have an options object rather than two params for the constructor call.
